pkg/metrics/metrics.go

```diff
@@ -144,6 +144,15 @@ The label 'result' can have the following values:
 		}, []string{"cluster_queue"},
 	)
 
+	quotaReservedPreemptionTime = prometheus.NewHistogramVec(
```

Comment:
I'm wondering if we could make this metric more generic, for example `evicted_workloads_time_seconds` and with labels `cluster_queue` and `reason`. Then `reason=Preempted` would carry the information you want, wdyt?

Note that we already have [`EvictedWorkloadsTotal`](https://github.com/kubernetes-sigs/kueue/blob/621787ee506ae9301afb4fa2305baf9296d794f5/pkg/metrics/metrics.go#L253) using this pattern.